### PR TITLE
Revert "feat: remove leader election for operator (#1000)"

### PR DESCRIPTION
**Pull request by zdtsw**
- we need leader election enabled for Operator
This reverts commit 3610b0ba0cf19bb32853d9d513c2f42ff8ab9f2d.
- cleanup manifests on configmap which is not used but still gets created by CSV from bundle

## How Has This Been Tested?

local build: quay.io/wenzhou/opendatahub-operator-catalog:v2.15.91
pod log:
```
I0805 13:12:45.727597       1 leaderelection.go:248] attempting to acquire leader lease openshift-operators/07ed84f7.opendatahub.io...
{"level":"info","ts":"2024-08-05T13:12:45Z","logger":"controller-runtime.certwatcher","msg":"Updated current TLS certificate"}
{"level":"info","ts":"2024-08-05T13:12:45Z","logger":"controller-runtime.webhook","msg":"Serving webhook server","host":"","port":9443}
{"level":"info","ts":"2024-08-05T13:12:45Z","logger":"controller-runtime.certwatcher","msg":"Starting certificate watcher"}
I0805 13:13:03.827022       1 leaderelection.go:258] successfully acquired lease openshift-operators/07ed84f7.opendatahub.io
```

## Merge criteria

- [ ] You have read the [contributors guide](https://github.com/opendatahub-io/opendatahub-operator/blob/incubation/CONTRIBUTING.md).
- [ ] Commit messages are meaningful - have a clear and concise summary and detailed explanation of what was changed and why.
- [ ] Pull Request contains a description of the solution, a link to the JIRA issue, and to any dependent or related Pull Request.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: config/manager/controller_manager_config.yaml
Comment (lburgazzoli) [inline config/manager/controller_manager_config.yaml:1]: is this file actually in use ? I don't see it mounted/referenced anywhere

Comment (zdtsw) [reply config/manager/controller_manager_config.yaml:1]: as the input to generate bundle/manifests/opendatahub-operator-manager-config_v1_configmap.yaml ?

Comment (zdtsw) [reply config/manager/controller_manager_config.yaml:1]: you are right.
we only copy over to bundle but we do not use it at all.
lets set it remove once for all!

File: bundle/manifests/opendatahub-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -1712,7 +1713,7 @@ spec:
                       fieldPath: metadata.namespace
                 - name: DEFAULT_MANIFESTS_PATH
                   value: /opt/manifests
-                image: REPLACE_IMAGE:latest
+                image: quay.io/wenzhou/opendatahub-operator:2.15.0
```
Comment (lburgazzoli) [inline bundle/manifests/opendatahub-operator.clusterserviceversion.yaml:1716]: is this expected ?

Comment (zdtsw) [reply bundle/manifests/opendatahub-operator.clusterserviceversion.yaml:1716]: will revert it once done testing after remove this CM

Comment (zdtsw) [reply bundle/manifests/opendatahub-operator.clusterserviceversion.yaml]: done

**Review by lburgazzoli — APPROVED** [review]

**Pull request merged**